api/src/modules/proxy/proxy.controller.ts

```diff
@@ -67,13 +66,98 @@ export class ProxyController {
     required: false,
     example: 'BRA.1',
   })
+  @Get('/administrative-areas/:level/preview/tiles/:z/:x/:y.mvt')
+  async proxyAdminAreaTile(@Req() request: Request, @Res() response: Response) {
+    return this.proxyService.proxyTileRequest(request, response);
+  }
+
+  @ApiOperation({
+    description: 'Get tile for protected areas.',
+  })
   /**
-   *
-   *@todo parse level from admin areas entity
-   *@todo add level validation here in proxy service. Duplicated on the geoprocessing api.
+   *@todo Change ApiOkResponse mvt type
    */
-  @Get('/administrative-areas/:level/preview/tiles/:z/:x/:y.mvt')
-  async proxyTile(@Req() request: Request, @Res() response: Response) {
+  @ApiOkResponse({
+    type: 'mvt',
+  })
+  @ApiUnauthorizedResponse()
+  @ApiForbiddenResponse()
+  @ApiParam({
+    name: 'z',
+    description: 'The zoom level ranging from 0 - 20',
+    type: Number,
+    required: true,
+  })
+  @ApiParam({
+    name: 'x',
+    description: 'The tile x offset on Mercator Projection',
+    type: Number,
+    required: true,
+  })
+  @ApiParam({
+    name: 'y',
+    description: 'The tile y offset on Mercator Projection',
+    type: Number,
+    required: true,
+  })
+  @ApiQuery({
+    name: 'wdpaid',
+    description: 'Id of WDPA area',
+    type: Number,
+    required: false,
+    example: '5555697751',
+  })
+  @Get('/protected-areas/preview/tiles/:z/:x/:y.mvt')
+  async proxyProtectedAreaTile(
+    @Req() request: Request,
+    @Res() response: Response,
+  ) {
+    return this.proxyService.proxyTileRequest(request, response);
+  }
+
+  @ApiOperation({
+    description: 'Get tile for a feature by id.',
+  })
+  /**
+   *@todo Change ApiOkResponse mvt type
+   */
+  @ApiOkResponse({
+    type: 'mvt',
+  })
+  @ApiUnauthorizedResponse()
+  @ApiForbiddenResponse()
+  @ApiParam({
+    name: 'z',
+    description: 'The zoom level ranging from 0 - 20',
+    type: Number,
+    required: true,
+  })
+  @ApiParam({
+    name: 'x',
+    description: 'The tile x offset on Mercator Projection',
+    type: Number,
+    required: true,
+  })
+  @ApiParam({
+    name: 'y',
+    description: 'The tile y offset on Mercator Projection',
+    type: Number,
+    required: true,
+  })
+  @ApiParam({
+    name: 'id',
+    description: 'Specific id of the feature',
+    type: String,
+    required: true,
+  })
+  @ApiQuery({
+    name: 'extentId',
+    description: 'Extent of the project',
+    type: Number,
+    required: false,
+  })
```

Comment:
is this correct? what does extentId map to (in our data model)?

Reply:
yes, this is what we defined on the [endpoints definition doc](https://docs.google.com/document/d/1T1Gp9RM6J-CoZoyIL7XBeUU1_w2ZrvN7rCSN_XAnBSg/edit#). If I remember well was the` features.id` but I cannot check on my local as the features import didn't work. @aagm could confirm

Reply:
This looks like a slip-on in my mind regarding how will we access the extent. I'm thinking if indeed we should refer to bbox here and filter by it.  But I think that based on the designs this is more related to the shape of the study area (or adm or extent, so I would refer it to a project id, and we can generate the specific logic from there.)